devito/types/tensor.py

```diff
@@ -391,16 +398,14 @@ class VectorTimeFunction(VectorFunction, TensorTimeFunction):
                  (False, True): TensorFunction, (False, False): VectorFunction}
 
 
-def vec_func(func1, func2):
-    f1 = getattr(func1, 'is_TimeDependent', False)
-    f2 = getattr(func2, 'is_TimeDependent', False)
-    return VectorTimeFunction if f1 or f2 else VectorFunction
+def vec_func(*funcs):
+    return (VectorTimeFunction if any([getattr(f, 'is_TimeDependent', False)
```

Comment:
nitpicking: if you have to split the line like that, you may as well make a multiline if-else 

```
if cond
   do A
else
   do B
```